roles/openshift_aws/tasks/vpc_and_subnet_id.yml

```diff
@@ -15,8 +15,14 @@
   ec2_vpc_subnet_facts:
     region: "{{ openshift_aws_region }}"
     filters:
-      "availability-zone": "{{ openshift_aws_subnet_az }}"
       vpc-id: "{{ vpcout.vpcs[0].id }}"
+      "tag:Name": "{{ item }}"
+  with_items: ['{%- if (openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list | length) == 1 -%}
+      {{ openshift_aws_subnet_az }}
+    {%- elif (openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list | length) > 1 -%}
+      {{ openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list }}
+    {%- else -%}
+    {%- endif -%}']
```

Comment:
could this be a filter plugin? It'd be better than this jinja stuff?

Reply:
TBH it would be nice to get rid of the "default subnet" code.  @kwoodson  @abutcher  hasn't given me a good explaination why "default subnet" is a better idea than just using element 0.  That would allow us to get rid of a lot of code including this bit.

Reply:
@joelddiaz you added the default_az stuff here https://github.com/openshift/openshift-ansible/pull/6603/files would you be ok with simply making element 0 the default az?

Reply:
@sdodson if the convention becomes that the first subnet in the list is treated as the default, that works for me. IIRC, my original purpose with #6603 was just to avoid having to set yet another role variable before running the provisioning.

Reply:
@mazzystr, I believe the intention was to be more explicit by defining a `default_az=True` rather than taking the first element in the list.  The initial implementation was using the first element in the list but wasn't explicit enough for the consumers so @joelddiaz added it. 

Now having said that, if you feel that going back to the original implementation is better then feel free.  

IMHO, I don't believe this method is any better than what is currently implemented.  Having a filter plugin is cleaner than jinja. An explicit declaration of the `default_az` is much better understood than an implicit understanding that element 0 is the default.